src/Test/StrictCheck/Shaped.hs

```diff
@@ -313,32 +313,34 @@ interleave p = unfold (fmap (project p)) . p
 -- @h % a@.
```

Comment:
I'm not sure I follow this generalization of `unzipWith`; can you elaborate on what's going on here? It seems strange to have the `(%)` type occur in the splitting function; the rest of these recursion schemes have a non-recursive operation as their argument, then extend that to the recursive (interleaved) case. Whereas this type indicates that the `split` function could deconstruct the results of the unzipping, kind of like a histomorphism. Is that what's intended here (and if so, what's the intended use)?

Reply:
Well the former type of the splitting function was `forall x . f x -> (g x, h x)`, but actually we know a bit more about `x`. I just put a typed hole were `split` was and took that as the new type of `split`. Now `split` has access to the already unzipped structure under the constructor that it splits right now. In my second commit I just changed the order of `split` and `crunch` (changing crunch a bit), as that yields a more useful type for split.

I originally did this, because I thought I could use the more general `zipWithM` to implement `entangleShape`, but it turned out not useful to be useful for that. Still it is strictly more general and the behaviour is the same.